airbyte-cdk/python/cdk-migrations.md

```diff
@@ -1,5 +1,36 @@
 # CDK Migration Guide
 
+## Upgrading to 5.0.0
+
+Version 5.0.0 of the CDK updates the `airbyte_cdk.models` dependency to replace Pydantic v2 models with Python `dataclasses`. It also
+updates the `airbyte-protocol-models` dependency to a version that uses dataclasses models.
+
+The changes to Airbyte CDK itself are backwards-compatible, but some changes are required if the connector:
+- uses the `airbyte_protocol` models directly, or `airbyte_cdk.models`, which points to `airbyte_protocol` models
```

Comment:
@artem1205 do you know if this will break any connectors?

If yes, I'm fine with upgrading to 5.0.0, but if not, I think it's preferable to stay on 4.X because of how difficult it still is to keep connectors on the latest major version.

Reply:
CAT tests were ✅ for source-:

- hardcoded-records
- zendesk-support
- shopify

These changes will not break anything, unless airbyte_protocol is imported directly (not as a part of `airbyte_cdk.models`)

I'll change the versioning to 4.x then.

Reply:
can you check if any connector directly import airbyte_protocol? It might be worth fixing them first to avoid the need to flag it as a breaking change

Reply:
With grep and excluding test folders got `18` connectors:
```bash
grep -r --include="*.py" --exclude-dir="unit_tests" --exclude-dir="integration_tests" "from airbyte_protocol" ./airbyte-integrations/connectors/
```


1. /source-linkedin-ads/
2. /source-shopify/
3. /source-amazon-ads/
4. /source-mixpanel/
5. /source-amazon-seller-partner/
6. /source-bing-ads/
7. /source-google-ads/
8. /source-salesforce/
9. /source-hardcoded-records/
10. /source-stripe/
11. /source-mailchimp/
12. /destination-snowflake-cortex/
13. /source-airtable/
14. /source-zendesk-support/
15. /source-marketo/
16. /destination-pinecone/
17. /source-pinterest/
18. /source-github/

Do you want me to fix all of them before merging this PR?